bc_obps/registration/api/user_operator.py

```diff
@@ -87,44 +87,74 @@ def save_operator(payload: UserOperatorOperatorIn, operator_instance: Operator,
         # create parent operator records
         operator_has_parent_operators: bool = payload.operator_has_parent_operators
         if operator_has_parent_operators:
-            po_operator_fields_mapping = {
-                "po_legal_name": "legal_name",
-                "po_trade_name": "trade_name",
-                "po_cra_business_number": "cra_business_number",
-                "po_bc_corporate_registry_number": "bc_corporate_registry_number",
-                "po_website": "website",
-            }
             for idx, po_operator in enumerate(payload.parent_operators_array):
-                new_po_operator_instance: ParentOperator = ParentOperator(
-                    child_operator=created_or_updated_operator_instance,
-                    operator_index=idx + 1,
+
+                existing_parent_operator_indices = list(
+                    ParentOperator.objects.filter(child_operator=created_or_updated_operator_instance).values_list(
+                        'operator_index', flat=True
+                    )
                 )
+
+                # archive any parent operators that have been removed
+                if existing_parent_operator_indices:
+
+                    def get_indices(po):
+                        return po.operator_index
+
+                    updated_parent_operator_indices = [
+                        get_indices(po) for po in payload.parent_operators_array if get_indices(po) is not None
+                    ]
+
+                    indices_to_delete = list(
+                        set(existing_parent_operator_indices) - set(updated_parent_operator_indices)
+                    )
+
+                    if indices_to_delete:
+                        for op_index in indices_to_delete:
+                            ParentOperator.objects.get(operator_index=op_index).set_archive(user.pk)
+
+                # assign an operator_index to new parent operators
+                if not po_operator.operator_index:
+                    highest_existing_index = (
+                        max(existing_parent_operator_indices) if len(existing_parent_operator_indices) > 0 else 0
+                    )
+
+                    po_operator.operator_index = highest_existing_index + 1
+
                 # handle addresses--if there's no mailing address given, it's the same as the physical address
                 po_physical_address = Address.objects.create(
                     street_address=po_operator.po_physical_street_address,
                     municipality=po_operator.po_physical_municipality,
                     province=po_operator.po_physical_province,
                     postal_code=po_operator.po_physical_postal_code,
                 )
```

Comment:
I suggest updating a parent operator's Address(physical/mailing) with new data instead of creating new Addresses (of course when we are doing the update). If we keep creating new Addresses, we may end up with many orphaned addresses that are not associated with any parent operator.

Reply:
Yes, I agree. There's a different ticket for that: https://github.com/orgs/bcgov/projects/122/views/2?pane=issue&itemId=51034863